src/schema.ts

```diff
@@ -1,25 +1,64 @@
-import { FileDescriptorProto } from 'ts-proto-descriptors';
+import {
+  DescriptorProto,
+  FileDescriptorProto,
+  FieldDescriptorProto,
+  FieldDescriptorProto_Type,
+} from 'ts-proto-descriptors';
 import { imp, code, Code, joinCode, def } from 'ts-poet';
 import { visit, visitServices } from './visit';
 import { Context } from './context';
 import SourceInfo from './sourceInfo';
 import { maybePrefixPackage } from './utils';
+import { basicTypeName, toReaderCall } from './types';
+import { Reader } from 'protobufjs/minimal';
 
 const fileDescriptorProto = imp('FileDescriptorProto@ts-proto-descriptors');
 
+const extensionCache: { [key: string]: FieldDescriptorProto } = {};
+
 export function generateSchema(ctx: Context, fileDesc: FileDescriptorProto, sourceInfo: SourceInfo): Code[] {
   const { options } = ctx;
   const chunks: Code[] = [];
 
+  fileDesc.extension.forEach((extension) => {
+    extensionCache[extension.number] = extension;
+  });
+
   chunks.push(code`
+    type ProtoMetaMessageOptions = {
```

Comment:
The reason why they get output outside of the existing object is because the existing objects get processed through `fromPartial` which would strip all of those fields from the runtime.